packages/e2e-test-utils-playwright/src/editor/index.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import type { Browser, Page, BrowserContext, Frame } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { clickBlockOptionsMenuItem } from './click-block-options-menu-item';
+import { clickBlockToolbarButton } from './click-block-toolbar-button';
+import { getEditedPostContent } from './get-edited-post-content';
+import { insertBlock } from './insert-block';
+import { openDocumentSettingsSidebar } from './open-document-settings-sidebar';
+import { openPreviewPage } from './preview';
+import { selectBlockByClientId } from './select-block-by-client-id';
+import { showBlockToolbar } from './show-block-toolbar';
+import { saveSiteEditorEntities } from './site-editor';
+
+type EditorConstructorProps = {
+	page: Page;
+	hasIframe?: boolean;
+};
+
+export class Editor {
```

Comment:
Asking each test to manually include `Editor` and initiate it in `test.use` might not be ergonomic for some. I wonder if we could just auto-initiate them in `test.ts` like we do for `pageUtils` and `admin`? The problem is with the `hasIframe` option though.

Apologies if I missed something here, but could we remove the `hasIframe` boolean, and just use the `canvas` getter for such cases? So that we can initiate the `editor` fixture globally and also allow accessing from the iframe canvas in each test.

Reply:
> Asking each test to manually include Editor and initiate it in test.use might not be ergonomic for some

Yeah, that is the downside.

> Apologies if I missed something here, but could we remove the hasIframe boolean, and just use the canvas getter for such cases?

The idea is that we can write `editor` utils that interact with the canvas and internally they can use `this.canvas`. The test configuration determines whether the iframe or the page is used. Removing `hasIframe` makes that difficult to do.

Potentially we could have some separate configuration that determines whether an iframe is used in a test util. For example, another class that just stores configuration? If it didn't rely on `Page` then it could be instantiated in `test.ts` and `hasIframe` could be set in a `beforeAll` only for site editor tests.

Reply:
I see. 🤔

Technically, I think we can still _useCanvas_ in `beforeEach` though? Would that make things easier?

Reply:
I think I managed to get it working without needing `Editor` to be setup in every test, the setup is now only needed in site editor tests. If this doesn't work we can try the beforeEach

Reply:
Cool! Amazingly simple solution 😆.